images/repmgr/agent/internal/pgconf/primaryconninfo.go

```
@@ -0,0 +1,131 @@
+// by PostgreSQL itself (atomically) on every ALTER SYSTEM, including #308's own
+// SetSynchronizedStandbySlots -- a plain truncating os.WriteFile here could interleave
+// with one of those and leave a half-written file, which is a config syntax error the
+// next postmaster start (or SIGHUP) refuses to parse. The temp file is created in the
+// same directory as path so the rename is same-filesystem (atomic, not a copy).
+func writeFileAtomic(path string, data []byte, perm os.FileMode) error {
+	tmp, err := os.CreateTemp(filepath.Dir(path), filepath.Base(path)+".tmp-*")
+	if err != nil {
+		return err
+	}
+	tmpPath := tmp.Name()
+	defer os.Remove(tmpPath) // no-op once the rename below succeeds
```

golangci-lint (2.12.2)

[error] Error return value of `os.Remove` is not checked

(errcheck)